File: 2-divide-y-conquista/2-binary-search/binary_search.py


def binary_search(array, n, a, b):

    if a > b:
        return -1
    
    m = (a + b) // 2

    if array[m] == n:
        return m

    if array[m] > n:
        return binary_search(array, n, a, m-1)
    
    return binary_search(array, n, m+1, b)


def search(array, n):
    return binary_search(array, n, 0, len(array) - 1)



array = [1, 2, 3, 4, 5, 6, 7]

print(search(array, 7))

File: 2-divide-y-conquista/2-binary-search/binarySearch.js


function binarySearch(array, n, a, b){

    if(a > b)
        return -1

    const m = (a + b) / 2;

    if(array[m] === n)
        return m

    if(array[m] > n)
        return binarySearch(array, n, a, m-1)

    return binarySearch(array, n, m+1, b)
}


function search(array, n){

    return binarySearch(array, n, 0, array.length - 1)
}